File: examples/sample.js
/**
 * Sample JavaScript File for GitHub Learning Repository
 * 
 * This is an example JavaScript file to demonstrate:
 * - Code structure
 * - Documentation
 * - Version control best practices
 */

/**
 * Generate a greeting message
 * @param {string} name - The name to greet
 * @returns {string} A greeting message
 */
function greet(name) {
    return `Hello, ${name}! Welcome to GitHub operations.`;
}

/**
 * Calculate the sum of two numbers
 * @param {number} a - First number
 * @param {number} b - Second number
 * @returns {number} The sum of a and b
 */
function calculateSum(a, b) {
    return a + b;
}

/**
 * Main function to demonstrate the script
 */
function main() {
    console.log(greet("GitHub Learner"));
    
    const result = calculateSum(5, 7);
    console.log(`The sum of 5 and 7 is: ${result}`);
    
    console.log("\n✅ Sample script executed successfully!");
}

// Run main function if this script is executed directly
if (require.main === module) {
    main();
}

// Export functions for use in other modules
module.exports = {
    greet,
    calculateSum,
    main
};

File: examples/sample.py
#!/usr/bin/env python3
"""
Sample Python Script for GitHub Learning Repository

This is an example Python file to demonstrate:
- Code structure
- Documentation
- Version control best practices
"""


def greet(name: str) -> str:
    """
    Generate a greeting message.
    
    Args:
        name: The name to greet
        
    Returns:
        A greeting message string
    """
    return f"Hello, {name}! Welcome to GitHub operations."


def calculate_sum(a: int, b: int) -> int:
    """
    Calculate the sum of two numbers.
    
    Args:
        a: First number
        b: Second number
        
    Returns:
        The sum of a and b
    """
    return a + b


def main():
    """Main function to demonstrate the script."""
    print(greet("GitHub Learner"))
    
    result = calculate_sum(5, 7)
    print(f"The sum of 5 and 7 is: {result}")
    
    print("\n✅ Sample script executed successfully!")


if __name__ == "__main__":
    main()
